Makefile

```diff
@@ -100,26 +164,36 @@ ifndef VERSION
 	$(error VERSION is undefined - run using make release VERSION=vX.Y.Z)
 endif
 	git tag $(VERSION)
-	$(MAKE) $(CONTAINER_NAME) 
+	$(MAKE) image
 	# Check that the version output appears on a line of its own (the -x option to grep).
 	# Tests that the "git tag" makes it into the binary. Main point is to catch "-dirty" builds
 	@echo "Checking if the tag made it into the binary"
-	docker run --rm calico/libnetwork-plugin$(ARCHTAG) -v | grep -x $(VERSION) || (echo "Reported version:" `docker run --rm calico/libnetwork-plugin$(ARCHTAG) -v` "\nExpected version: $(VERSION)" && exit 1)
-	docker tag calico/libnetwork-plugin$(ARCHTAG) calico/libnetwork-plugin$(ARCHTAG):$(VERSION)
-	docker tag calico/libnetwork-plugin$(ARCHTAG) quay.io/calico/libnetwork-plugin$(ARCHTAG):$(VERSION)
-	docker tag calico/libnetwork-plugin$(ARCHTAG) quay.io/calico/libnetwork-plugin$(ARCHTAG):latest
-
-	@echo "Now push the tag and images. Then create a release on Github and attach the $(DIST)/libnetwork-plugin binary"
+	docker run --rm $(CONTAINER_NAME) -v | grep -x $(VERSION) || (echo "Reported version:" `docker run --rm $(CONTAINER_NAME) -v` "\nExpected version: $(VERSION)" && exit 1)
+ifeq ($(ARCH),amd64)
+	docker tag $(CONTAINER_NAME) calico/libnetwork-plugin:$(VERSION)
+	docker tag $(CONTAINER_NAME) quay.io/calico/libnetwork-plugin:$(VERSION)
+	docker tag $(CONTAINER_NAME) quay.io/calico/libnetwork-plugin:latest
+endif
+	docker tag $(CONTAINER_NAME) calico/libnetwork-plugin:$(VERSION)-$(ARCH)
+	docker tag $(CONTAINER_NAME) quay.io/calico/libnetwork-plugin:$(VERSION)-$(ARCH)
+	docker tag $(CONTAINER_NAME) quay.io/calico/libnetwork-plugin:latest-$(ARCH)
+	@echo "Now push the tag and images. Then create a release on Github and attach the dist/libnetwork-plugin binary"
 	@echo "git push origin $(VERSION)"
-	@echo "docker push calico/libnetwork-plugin$(ARCHTAG):$(VERSION)"
-	@echo "docker push quay.io/calico/libnetwork-plugin$(ARCHTAG):$(VERSION)"
-	@echo "docker push calico/libnetwork-plugin$(ARCHTAG):latest"
-	@echo "docker push quay.io/calico/libnetwork-plugin$(ARCHTAG):latest"
+ifeq ($(ARCH),amd64)
+	@echo "docker push calico/libnetwork-plugin:$(VERSION)"
+	@echo "docker push quay.io/calico/libnetwork-plugin:$(VERSION)"
+	@echo "docker push calico/libnetwork-plugin:latest"
+	@echo "docker push quay.io/calico/libnetwork-plugin:latest"
+endif
+	@echo "docker push calico/libnetwork-plugin:$(VERSION)-$(ARCH)"
+	@echo "docker push quay.io/calico/libnetwork-plugin:$(VERSION)-$(ARCH)"
+	@echo "docker push calico/libnetwork-plugin:latest-$(ARCH)"
+	@echo "docker push quay.io/calico/libnetwork-plugin:latest-$(ARCH)"
```

Comment:
can we create a target here which pushes images and mention that make target here so that it will become easy whovewer releasing them? someting like `make push`

Reply:
Adding a push target that that will actually do the push sounds good to me,  but as I wont be doing the release I would like input from the maintainer.  @robbrockbank

Reply:
I had a long discussion on Slack with @fasaxc about target `push`. As I understand it, there is a define desire to do a `push` target.

I captured the desired semantics as I understood them in [felix PR](https://github.com/projectcalico/felix/pull/1746) but it hasn't been merged yet, so still might change. Specifically, look [here](https://github.com/projectcalico/felix/pull/1746/files#diff-b67911656ef5d18c4ae36cb6741b7965R304)

```Makefile
push: image imagetag
	docker tag calico/felix:latest-$(ARCH) calico/felix:$(IMAGETAG)-$(ARCH)
	docker tag calico/felix:latest-$(ARCH) quay.io/calico/felix:$(IMAGETAG)-$(ARCH)
	docker push calico/felix:$(IMAGETAG)-$(ARCH)
	docker push quay.io/calico/felix:$(IMAGETAG)-$(ARCH)
ifeq ($(ARCH),amd64)
	docker tag calico/felix:latest-$(ARCH) calico/felix:$(IMAGETAG)
	docker tag calico/felix:latest-$(ARCH) quay.io/calico/felix:$(IMAGETAG)
	docker push calico/felix:$(IMAGETAG)
	docker push quay.io/calico/felix:$(IMAGETAG)
endif
```

Basically `make push`:

* checks that `IMAGETAG` is defined
* tags and pushes the current `:IMAGETAG-ARCH` for docker hub and quay.io
* if `amd64`, tags and pushes out the archless `:IMAGETAG` for docker hub and quay.io

The last step will be changed for multi-arch manifest if/when product management decide it is good.

Reply:
Sorry to drive you a little crazy with the changes. We now have the (modified) version of how `push` etc. work in `felix`, `kube-controllers`, etc. I will reference below.